plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryResultPageSource.java

```diff
@@ -206,8 +214,10 @@ private static void writeSlice(BlockBuilder output, Type type, Object value)
             type.writeSlice(output, utf8Slice(((Utf8) value).toString()));
         }
         else if (type instanceof DecimalType) {
-            BigDecimal bdValue = DECIMAL_CONVERTER.convert(value);
-            type.writeSlice(output, Decimals.encodeScaledValue(bdValue, NUMERIC_DATA_TYPE_SCALE));
+            verify(isLongDecimal(type));
```

Comment:
Is this assumption true because short decimals are encoded as `long` and not `Slice`? So `writeSlice` won't be called for short decimals?

Reply:
Yes, this won't be called for short decimals. In case of short decimals, `type.writeLong(...)` would be used at L174.